tutorials/javaFxPart5.md

```diff
@@ -0,0 +1,419 @@
+     // current code...
+     dialog.getStyleClass().add("reply-label");
+ }
+```
+
+## Padding and Margin
+
+In short, padding represents the space between the content and its border, while margin represents the invisible space
+around the border.
+
+![Padding and Margin](images/javafx/PaddingAndMargin.png)
+
+Let us play around with the padding and margin of the text bubble.
+
+Open `DialogBox.fxml` in SceneBuilder. Click on `Label`, and look at layout in the right accordion panel.
+
+![Padding and Margin Demo](images/javafx/PaddingAndMarginDemo.png)
+
+Play around with the padding and margin and find the parameters that is most suitable for you! To better view the
+effects of the margin, you may either
+- Run the app
+- Select both image and label to view the distance between them
+  ![View effects of margin](images/javafx/ViewEffectsOfMargin.png)
+
+## Borders
+
+Labels, TextBoxes, Buttons, more or less all the nodes used so far are subclasses of
+[Region](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html#region), and hence share the following
+CSS properties for borders:
+
+| CSS property      | Description                                          | 
+|-------------------|------------------------------------------------------|
+| `-fx-border-color`  | Set border color.                                    |
+| `-fx-border-radius` | Set the "roundedness" of the border.                 |
+| `-fx-border-width`  | Set border thickness.                                |
+| `-fx-border-style`  | Set style of border (e.g., can be dashed or bolded). |
+| `-fx-border-insets` | Set padding between the content and the border.      |
+
+For _all_ of these properties, you can either set a single value to apply to all corners/sides, or you can set them
+individually on each corner/side.
+
+To demonstrate this, let us take a look at the following code for our text bubble:
+
+**dialog-box.css**
+```css
+.label {
+    ...
+    -fx-border-color: #d55e00 #009e73 #cc79a7 #0072b2;
+    ...
+    -fx-background-radius: 1em 1em 0 1em;
+    -fx-border-radius: 1em 1em 0 1em;
+}
+```
+
+The values are set in a **clockwise** manner starting with the **top left corner/top edge**.
+
+![BorderDemo](images/javafx/BorderDemo.png)
+
+## Color
+
+There are a few ways to set a color.
+1. Using `rgb(...)` / `rgba(...)` (use the fourth parameter to set opacity from 0 to 1)
+2. Using HEX (`#ffffff`)
+3. Using an existing [named color](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html#typecolor)
+4. Using your own defined (looked-up) color
+
+### Looked-up Color
+You can define a color within the `.root` selector, which can be referenced by other classes
+within the stylesheet.
+
+**main.css**
+```css
+.root {
+    duke: rgb(237, 255, 242);     /* Create a looked-up color called "duke" within root */
+    -fx-background-color: duke;
+}
+
+.scroll-bar {
+    ...
+    -fx-background-color: duke;   /* Using looked-up color */
+}
+```
```

Comment:
Use a meaningful name for the color?

Reply:
Hmmm, can't really think of an appropriate name, `themecolor` maybe?